content/sensu-core/0.29/guides/pre-compile-plugins.md

```diff
@@ -0,0 +1,142 @@
+_NOTE: Compiled gems are tied to CPU architecture for which they are compiled on._
+
+## Installing The Build Toolchain
+
+We'll first add the Sensu yum repository definition and install Sensu
+
+{{< highlight shell >}}
+printf '[sensu]
+name=sensu
+baseurl=https://repositories.sensuapp.org/yum/$releasever/$basearch/
+gpgcheck=0
+enabled=1' | sudo tee /etc/yum.repos.d/sensu.repo && sudo yum install -y sensu{{< /highlight >}}
+
+And then install EPEL, Dev tools and libs required for the next steps
+
+{{< highlight shell >}}
+sudo yum install -y epel-release
+sudo yum groupinstall -y "Development Tools
+# installing dependencies for nokogiri
+sudo yum install -y libxml2 libxslt libxslt-devel libxml2-devel"{{< /highlight >}}
+
+Finally we'll add Sensu's embedded bin to our path
+
+{{< highlight shell >}}
+export PATH=$PATH:/opt/sensu/embedded/bin{{< /highlight >}}
+
+## Installing a Gem to a tmp directory
+
+Install the gem-compiler package
+
+{{< highlight shell >}}
+gem install gem-compiler{{< /highlight >}}
+
+We'll now install the source gems into a cache directory
+
+{{< highlight shell >}}
+mkdir /tmp/gems
+gem install --no-ri --no-rdoc --install-dir /tmp/gems sensu-plugins-aws --version '=12.3.0'
+gem install --no-ri --no-rdoc --install-dir /tmp/gems mini_portile2{{< /highlight >}}
+
+Observing the `gem install` output, we can see that nokogiri and unf_ext both build native extensions.
+These are the gems we will compile so we can install them from our own repository.
+
+### Compile the Gems for packaging
+
+We'll use `gem-compiler` to rebuild the gem packages with the included native extensions:
+
+{{< highlight shell >}}
+cd /tmp/gems/cache
+gem compile unf_ext-0.0.7.5.gem
+gem compile nokogiri-1.8.4.gem --prune{{< /highlight >}}
+
+_NOTE: `nokogiri` requires --prune flag per gem-compiler readme notes _
+
+The resulting compiled gem files have a x86_64-linux in their file name denoting their system architech:
+
+{{< highlight shell >}}
+$ ls  -1 *-x86_64-linux.gem
+nokogiri-1.8.1-x86_64-linux.gem
+unf_ext-0.0.7.4-x86_64-linux.gem{{< /highlight >}}
+
+## Install Pre-Compiled Gems on Targeted System and Testing
+
+### Install compiled gem into embedded Ruby
+Now we can copy these gems to another system, install them into the sensu embedded ruby. 
+In a production or development enviroment, you'd want to put these in a rubygems server of your own (e.g. [Geminabox(https://github.com/geminabox/geminabox), [Artifactory](https://jfrog.com/artifactory/) or similar). You can also use a tool like `fpm` to convert the gems into a system package.
+
+Here we've copied the compiled gems to a system with Sensu installed, but no gcc or other compile toolchain
+
+ _NOTE: nokogiri needs libxml2 and possibly libxslt to be present on your system._
+
+{{< highlight shell >}}
+$ /opt/sensu/embedded/bin/gem install unf_ext-*-x86_64-linux.gem
+$ /opt/sensu/embedded/bin/gem install nokogiri-*-x86_64-linux.gem{{< /highlight >}}
```

Comment:
I get error here. I had to use the absolute path for `gem` though as `export` does not get to `sudo` (see comment above. That said, here is the error:

```
[vagrant@pluygintest cache]$ /opt/sensu/embedded/bin/gem compile nokogiri-1.8.4.gem --prune
Unpacking gem: 'nokogiri-1.8.4' in temporary directory...
ERROR:  While executing gem ... (Gem::InstallError)
    invalid gem: No such file or directory @ rb_sysopen - nokogiri-1.8.4.gem
```